packages/grafana-ui/src/components/Typeahead/Typeahead.tsx

```diff
@@ -31,21 +30,28 @@ interface State {
 }
 
 export class Typeahead extends React.PureComponent<Props, State> {
+  static contextType = ThemeContext;
+  context!: React.ContextType<typeof ThemeContext>;
   listRef = createRef<FixedSizeList>();
 
-  constructor(props: Props) {
-    super(props);
-
-    const allItems = flattenGroupItems(props.groupedItems);
-    const longestLabel = calculateLongestLabel(allItems);
-    const { listWidth, listHeight, itemHeight } = calculateListSizes(props.theme, allItems, longestLabel);
-    this.state = { listWidth, listHeight, itemHeight, hoveredItem: null, typeaheadIndex: 1, allItems };
-  }
+  state: State = { hoveredItem: -1, typeaheadIndex: 1, allItems: [], listWidth: -1, listHeight: -1, itemHeight: -1 };
```

Comment:
using -1 for an unset `hoveredItem` will not work with its use of conditional logic in `render`

Reply:
ah, thanks for that catch, will fix